### ISSUE TITLE
Circuit Breaker: Too many dynamic script compilations within one minute

### ISSUE THREAD

**Issue by aleph-zero** [elastic/elasticsearch#26553]
**Elasticsearch version** (`bin/elasticsearch --version`):
6.0.0-beta2

**Plugins installed**: []
ingest-geoip
x-pack

**JVM version** (`java -version`):
java version "1.8.0_111"
Java(TM) SE Runtime Environment (build 1.8.0_111-b14)
Java HotSpot(TM) 64-Bit Server VM (build 25.111-b14, mixed mode)

**OS version** (`uname -a` if on a Unix-like system):
Darwin disciple.local 16.7.0 Darwin Kernel Version 16.7.0: Thu Jun 15 17:36:27 PDT 2017; root:xnu-3789.70.16~2/RELEASE_X86_64 x86_64

**Description of the problem including expected versus actual behavior**:
Upon startup of a fresh install of 6.0.0-beta2, I see error messages in the logs about ```too many script compilations``` (see logs below).

**Steps to reproduce**:

Please include a *minimal* but *complete* recreation of the problem, including
(e.g.) index creation, mappings, settings, query etc.  The easier you make for
us to reproduce it, the more likely that somebody will take the time to look at it.

 1. Download and install 6.0.0-beta2 version of: filebeat, metricbeat, x-pack, kibana, elasticsearch
 2. Install ingest-geoip
 3. Start elasticsearch, kibana, filebeat and metricbeat. Everything OK.
 4. Stop elasticsearch. 
 5. Edit jvm.options to -Xms=2g -Xmx=2g
 6. Start elasticsearch (filebeat, metricbeat, and kibana are still running from step 2)
 7. Observer circuit breaker log messages in elasticsearch.log  

**Provide logs (if relevant)**:

```
[2017-09-08T10:36:01,332][INFO ][o.e.g.GatewayService     ] [Af_l2wc] recovered [10] indices into cluster_state
[2017-09-08T10:36:01,806][INFO ][o.e.c.r.a.AllocationService] [Af_l2wc] Cluster health status changed from [RED] to [YELLOW] (reason: [shards started [[.watcher-history-6-2017.09.08][0]] ...]).
[2017-09-08T10:36:02,182][ERROR][o.e.x.w.WatcherService   ] [Af_l2wc] couldn't load watch [i6rfX2x4SDuQbJ3bnrA8Zw_elasticsearch_cluster_status], ignoring it...
org.elasticsearch.script.GeneralScriptException: Failed to compile inline script [ctx.vars.fails_check = ctx.payload.check.hits.total != 0 && ctx.payload.check.hits.hits[0]._source.cluster_state.status != 'green';ctx.vars.not_resolved = ctx.payload.alert.hits.total == 1 && ctx.payload.alert.hits.hits[0]._source.resolved_timestamp == null;return ctx.vars.fails_check || ctx.vars.not_resolved] using lang [painless]
	at org.elasticsearch.script.ScriptService.compile(ScriptService.java:301) ~[elasticsearch-6.0.0-beta2.jar:6.0.0-beta2]
	at org.elasticsearch.xpack.watcher.condition.ScriptCondition.<init>(ScriptCondition.java:58) ~[?:?]
	at org.elasticsearch.xpack.watcher.condition.ScriptCondition.parse(ScriptCondition.java:68) ~[?:?]
	at org.elasticsearch.xpack.watcher.Watcher.lambda$createComponents$4(Watcher.java:255) ~[?:?]
	at org.elasticsearch.xpack.watcher.condition.ConditionRegistry.parseExecutable(ConditionRegistry.java:68) ~[?:?]
	at org.elasticsearch.xpack.watcher.watch.Watch$Parser.parse(Watch.java:300) ~[?:?]
	at org.elasticsearch.xpack.watcher.watch.Watch$Parser.parse(Watch.java:266) ~[?:?]
	at org.elasticsearch.xpack.watcher.watch.Watch$Parser.parse(Watch.java:231) ~[?:?]
	at org.elasticsearch.xpack.watcher.WatcherService.loadWatches(WatcherService.java:284) ~[?:?]
	at org.elasticsearch.xpack.watcher.WatcherService.start(WatcherService.java:138) ~[?:?]
	at org.elasticsearch.xpack.watcher.WatcherLifeCycleService.start(WatcherLifeCycleService.java:95) ~[?:?]
	at org.elasticsearch.xpack.watcher.WatcherLifeCycleService.lambda$clusterChanged$4(WatcherLifeCycleService.java:178) ~[?:?]
	at org.elasticsearch.common.util.concurrent.ThreadContext$ContextPreservingRunnable.run(ThreadContext.java:569) [elasticsearch-6.0.0-beta2.jar:6.0.0-beta2]
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142) [?:1.8.0_111]
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617) [?:1.8.0_111]
	at java.lang.Thread.run(Thread.java:745) [?:1.8.0_111]
Caused by: org.elasticsearch.common.breaker.CircuitBreakingException: [script] Too many dynamic script compilations within one minute, max: [15/min]; please use on-disk, indexed, or scripts with parameters instead; this limit can be changed by the [script.max_compilations_per_minute] setting
	at org.elasticsearch.script.ScriptService.checkCompilationLimit(ScriptService.java:342) ~[elasticsearch-6.0.0-beta2.jar:6.0.0-beta2]
	at org.elasticsearch.script.ScriptService.compile(ScriptService.java:295) ~[elasticsearch-6.0.0-beta2.jar:6.0.0-beta2]
	... 15 more
[2017-09-08T10:36:06,554][ERROR][o.e.x.m.e.l.LocalExporter] failed to get monitoring watch [i6rfX2x4SDuQbJ3bnrA8Zw_elasticsearch_cluster_status]
org.elasticsearch.script.GeneralScriptException: Failed to compile inline script [ctx.vars.email_recipient = (ctx.payload.kibana_settings.hits.total > 0) ? ctx.payload.kibana_settings.hits.hits[0]._source.kibana_settings.xpack.default_admin_email : null;ctx.vars.is_new = ctx.vars.fails_check && !ctx.vars.not_resolved;ctx.vars.is_resolved = !ctx.vars.fails_check && ctx.vars.not_resolved;def state = ctx.payload.check.hits.hits[0]._source.cluster_state.status;if (ctx.vars.not_resolved){ctx.payload = ctx.payload.alert.hits.hits[0]._source;if (ctx.vars.fails_check == false) {ctx.payload.resolved_timestamp = ctx.execution_time;}} else {ctx.payload = ['timestamp': ctx.execution_time, 'metadata': ctx.metadata.xpack];}if (ctx.vars.fails_check) {ctx.payload.prefix = 'Elasticsearch cluster status is ' + state + '.';if (state == 'red') {ctx.payload.message = 'Allocate missing primary shards and replica shards.';ctx.payload.metadata.severity = 2100;} else {ctx.payload.message = 'Allocate missing replica shards.';ctx.payload.metadata.severity = 1100;}}ctx.vars.state = state.toUpperCase();ctx.payload.update_timestamp = ctx.execution_time;return ctx.payload;] using lang [painless]
	at org.elasticsearch.script.ScriptService.compile(ScriptService.java:301) ~[elasticsearch-6.0.0-beta2.jar:6.0.0-beta2]
	at org.elasticsearch.xpack.watcher.transform.script.ExecutableScriptTransform.<init>(ExecutableScriptTransform.java:47) ~[?:?]
	at org.elasticsearch.xpack.watcher.transform.script.ScriptTransformFactory.createExecutable(ScriptTransformFactory.java:49) ~[?:?]
	at org.elasticsearch.xpack.watcher.transform.script.ScriptTransformFactory.createExecutable(ScriptTransformFactory.java:28) ~[?:?]
	at org.elasticsearch.xpack.watcher.transform.TransformFactory.parseExecutable(TransformFactory.java:53) ~[?:?]
	at org.elasticsearch.xpack.watcher.transform.TransformRegistry.parse(TransformRegistry.java:64) ~[?:?]
	at org.elasticsearch.xpack.watcher.transform.TransformRegistry.parse(TransformRegistry.java:53) ~[?:?]
	at org.elasticsearch.xpack.watcher.watch.Watch$Parser.parse(Watch.java:302) ~[?:?]
	at org.elasticsearch.xpack.watcher.watch.Watch$Parser.parse(Watch.java:266) ~[?:?]
	at org.elasticsearch.xpack.watcher.watch.Watch$Parser.parseWithSecrets(Watch.java:252) ~[?:?]
	at org.elasticsearch.xpack.watcher.transport.actions.get.TransportGetWatchAction.lambda$masterOperation$0(TransportGetWatchAction.java:77) ~[?:?]
	at org.elasticsearch.action.ActionListener$1.onResponse(ActionListener.java:59) ~[elasticsearch-6.0.0-beta2.jar:6.0.0-beta2]
	at org.elasticsearch.action.support.ContextPreservingActionListener.onResponse(ContextPreservingActionListener.java:43) ~[elasticsearch-6.0.0-beta2.jar:6.0.0-beta2]
	at org.elasticsearch.action.support.TransportAction$1.onResponse(TransportAction.java:85) ~[elasticsearch-6.0.0-beta2.jar:6.0.0-beta2]
	at org.elasticsearch.action.support.TransportAction$1.onResponse(TransportAction.java:81) ~[elasticsearch-6.0.0-beta2.jar:6.0.0-beta2]
	at org.elasticsearch.action.support.ContextPreservingActionListener.onResponse(ContextPreservingActionListener.java:43) ~[elasticsearch-6.0.0-beta2.jar:6.0.0-beta2]
	at org.elasticsearch.action.support.single.shard.TransportSingleShardAction$AsyncSingleAction$2.handleResponse(TransportSingleShardAction.java:247) ~[elasticsearch-6.0.0-beta2.jar:6.0.0-beta2]
	at org.elasticsearch.action.support.single.shard.TransportSingleShardAction$AsyncSingleAction$2.handleResponse(TransportSingleShardAction.java:233) ~[elasticsearch-6.0.0-beta2.jar:6.0.0-beta2]
	at org.elasticsearch.transport.TransportService$ContextRestoreResponseHandler.handleResponse(TransportService.java:1053) ~[elasticsearch-6.0.0-beta2.jar:6.0.0-beta2]
	at org.elasticsearch.transport.TransportService$DirectResponseChannel.processResponse(TransportService.java:1127) ~[elasticsearch-6.0.0-beta2.jar:6.0.0-beta2]
	at org.elasticsearch.transport.TransportService$DirectResponseChannel.sendResponse(TransportService.java:1117) ~[elasticsearch-6.0.0-beta2.jar:6.0.0-beta2]
	at org.elasticsearch.transport.TransportService$DirectResponseChannel.sendResponse(TransportService.java:1106) ~[elasticsearch-6.0.0-beta2.jar:6.0.0-beta2]
	at org.elasticsearch.transport.DelegatingTransportChannel.sendResponse(DelegatingTransportChannel.java:60) ~[elasticsearch-6.0.0-beta2.jar:6.0.0-beta2]
	at org.elasticsearch.transport.RequestHandlerRegistry$TransportChannelWrapper.sendResponse(RequestHandlerRegistry.java:108) ~[elasticsearch-6.0.0-beta2.jar:6.0.0-beta2]
	at org.elasticsearch.action.support.single.shard.TransportSingleShardAction$ShardTransportHandler.messageReceived(TransportSingleShardAction.java:295) ~[elasticsearch-6.0.0-beta2.jar:6.0.0-beta2]
	at org.elasticsearch.action.support.single.shard.TransportSingleShardAction$ShardTransportHandler.messageReceived(TransportSingleShardAction.java:287) ~[elasticsearch-6.0.0-beta2.jar:6.0.0-beta2]
	at org.elasticsearch.transport.TransportRequestHandler.messageReceived(TransportRequestHandler.java:30) ~[elasticsearch-6.0.0-beta2.jar:6.0.0-beta2]
	at org.elasticsearch.xpack.security.transport.SecurityServerTransportInterceptor$ProfileSecuredRequestHandler$1.doRun(SecurityServerTransportInterceptor.java:261) ~[?:?]
	at org.elasticsearch.common.util.concurrent.AbstractRunnable.run(AbstractRunnable.java:37) ~[elasticsearch-6.0.0-beta2.jar:6.0.0-beta2]
	at org.elasticsearch.common.util.concurrent.EsExecutors$1.execute(EsExecutors.java:139) ~[elasticsearch-6.0.0-beta2.jar:6.0.0-beta2]
	at org.elasticsearch.xpack.security.transport.SecurityServerTransportInterceptor$ProfileSecuredRequestHandler.lambda$messageReceived$0(SecurityServerTransportInterceptor.java:310) ~[?:?]
	at org.elasticsearch.action.ActionListener$1.onResponse(ActionListener.java:59) ~[elasticsearch-6.0.0-beta2.jar:6.0.0-beta2]
	at org.elasticsearch.xpack.security.transport.ServerTransportFilter$NodeProfile.lambda$null$2(ServerTransportFilter.java:158) ~[?:?]
	at org.elasticsearch.xpack.security.authz.AuthorizationUtils$AsyncAuthorizer.maybeRun(AuthorizationUtils.java:124) ~[?:?]
	at org.elasticsearch.xpack.security.authz.AuthorizationUtils$AsyncAuthorizer.setRunAsRoles(AuthorizationUtils.java:118) ~[?:?]
	at org.elasticsearch.xpack.security.authz.AuthorizationUtils$AsyncAuthorizer.authorize(AuthorizationUtils.java:106) ~[?:?]
	at org.elasticsearch.xpack.security.transport.ServerTransportFilter$NodeProfile.lambda$inbound$3(ServerTransportFilter.java:160) ~[?:?]
	at org.elasticsearch.action.ActionListener$1.onResponse(ActionListener.java:59) ~[elasticsearch-6.0.0-beta2.jar:6.0.0-beta2]
	at org.elasticsearch.xpack.security.authc.AuthenticationService$Authenticator.lambda$authenticateAsync$2(AuthenticationService.java:195) ~[x-pack-6.0.0-beta2.jar:6.0.0-beta2]
	at org.elasticsearch.xpack.security.authc.AuthenticationService$Authenticator.lambda$lookForExistingAuthentication$4(AuthenticationService.java:228) ~[x-pack-6.0.0-beta2.jar:6.0.0-beta2]
	at org.elasticsearch.xpack.security.authc.AuthenticationService$Authenticator.lookForExistingAuthentication(AuthenticationService.java:239) [x-pack-6.0.0-beta2.jar:6.0.0-beta2]
	at org.elasticsearch.xpack.security.authc.AuthenticationService$Authenticator.authenticateAsync(AuthenticationService.java:193) [x-pack-6.0.0-beta2.jar:6.0.0-beta2]
	at org.elasticsearch.xpack.security.authc.AuthenticationService$Authenticator.access$000(AuthenticationService.java:147) [x-pack-6.0.0-beta2.jar:6.0.0-beta2]
	at org.elasticsearch.xpack.security.authc.AuthenticationService.authenticate(AuthenticationService.java:116) [x-pack-6.0.0-beta2.jar:6.0.0-beta2]
	at org.elasticsearch.xpack.security.transport.ServerTransportFilter$NodeProfile.inbound(ServerTransportFilter.java:139) [x-pack-6.0.0-beta2.jar:6.0.0-beta2]
	at org.elasticsearch.xpack.security.transport.SecurityServerTransportInterceptor$ProfileSecuredRequestHandler.messageReceived(SecurityServerTransportInterceptor.java:317) [x-pack-6.0.0-beta2.jar:6.0.0-beta2]
	at org.elasticsearch.transport.RequestHandlerRegistry.processMessageReceived(RequestHandlerRegistry.java:66) [elasticsearch-6.0.0-beta2.jar:6.0.0-beta2]
	at org.elasticsearch.transport.TransportService$7.doRun(TransportService.java:650) [elasticsearch-6.0.0-beta2.jar:6.0.0-beta2]
	at org.elasticsearch.common.util.concurrent.ThreadContext$ContextPreservingAbstractRunnable.doRun(ThreadContext.java:638) [elasticsearch-6.0.0-beta2.jar:6.0.0-beta2]
	at org.elasticsearch.common.util.concurrent.AbstractRunnable.run(AbstractRunnable.java:37) [elasticsearch-6.0.0-beta2.jar:6.0.0-beta2]
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142) [?:1.8.0_111]
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617) [?:1.8.0_111]
	at java.lang.Thread.run(Thread.java:745) [?:1.8.0_111]
Caused by: org.elasticsearch.common.breaker.CircuitBreakingException: [script] Too many dynamic script compilations within one minute, max: [15/min]; please use on-disk, indexed, or scripts with parameters instead; this limit can be changed by the [script.max_compilations_per_minute] setting
	at org.elasticsearch.script.ScriptService.checkCompilationLimit(ScriptService.java:342) ~[elasticsearch-6.0.0-beta2.jar:6.0.0-beta2]
	at org.elasticsearch.script.ScriptService.compile(ScriptService.java:295) ~[elasticsearch-6.0.0-beta2.jar:6.0.0-beta2]
```

**Comment by rfraposa** [comment]
I am having the same issue. When I restart Elasticsearch 6.0.0-beta2 with X-Pack installed, I get a handful of `Too many dynamic script compilations within one minute` exceptions, from both Security and Watcher scripts.

**Comment by jasontedor** [comment]
This is fixed by #26399 and will appear in a future pre-release. Also, please do not report X-Pack issues here. This repository is for open source Elasticsearch only.

**Comment by mohitboston** [comment]
Is this issue also in version 5.2? We are seeing this
